qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -313,6 +322,217 @@ def _run(self, qobj, job_id=''):
+
+    def _inline_circuit(self, circ, continue_label, break_label):
+        ret = circ.copy()
+        ret.data = []
+
+        q2i = {}
+        for q in ret.qubits:
+            q2i[q] = len(q2i)
+        c2i = {}
+        for c in ret.clbits:
+            c2i[c] = len(c2i)
+
+        for inst, qargs, cargs in circ.data:
+            if isinstance(inst, ForLoopOp):
+                ret.append(self._inline_for_loop_op(inst),
+                           [q2i[q] for q in qargs],
+                           [c2i[c] for c in cargs])
+            elif isinstance(inst, WhileLoopOp):
+                (cond_tuple, continue_label,
+                 loop_start_label, break_label, body) = self._inline_while_loop_op(inst)
+                c_if_args = self._convert_c_if_args(cond_tuple)
+                ret.append(AerMark(continue_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+                ret.append(AerJump(loop_start_label, ret.num_qubits).c_if(*c_if_args),
+                           range(ret.num_qubits), [])
+                ret.append(AerJump(break_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+                ret.append(AerMark(loop_start_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+                ret.append(body, [q2i[q] for q in qargs], [c2i[c] for c in cargs])
+            elif isinstance(inst, IfElseOp):
+                cond_tuple, true_label, else_label, body = self._inline_if_else_op(inst,
+                                                                                   continue_label,
+                                                                                   break_label)
+                c_if_args = self._convert_c_if_args(cond_tuple)
+                ret.append(AerJump(true_label, ret.num_qubits).c_if(*c_if_args),
+                           range(ret.num_qubits), [])
+                ret.append(AerJump(else_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+                ret.append(body, [q2i[q] for q in qargs], [c2i[c] for c in cargs])
+            elif isinstance(inst, BreakLoopOp):
+                ret.append(AerJump(break_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+            elif isinstance(inst, ContinueLoopOp):
+                ret.append(AerJump(continue_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+            else:
+                ret.append(inst, qargs, cargs)
+
+        return ret
+
+    def _compile(self, circuits):
+        if isinstance(circuits, list):
+            all_static = all([not self._is_dynamic(circuit) for circuit in circuits])
+            if all_static:
+                return circuits
+            basis_gates = self.configuration().basis_gates + ['mark', 'jump']
+            return [transpile(self._inline_circuit(self._decompose_subcircuits(circuit),
```

Comment:
This function is very difficult to read. It would be a lot clearer if it was something like:
```python
if not isinstance(circuits, list):
    circuits = [circuits]

compiled_circuits = []
for circuit in circuits:
    if not _is_dynamic(circuit):
         compiled_circuits.append(circuit)
         continue

    decomposed_circuit = _decompose_subcircuits(circuit)
    compiled_circuits.append(_inline_circuit(decomposed_circuit, None, None))

return compiled_circuits
```